README.md

```diff
@@ -40,364 +43,346 @@ some kind of discriminant:
-  has(E: object, key: string | symbol): boolean;
-  hasValue(E: object, value: any): boolean;
-  getName(E: object, value: any): string | undefined;
-  format(E: object, value: any): string | symbol | undefined;
-  parse(E: object, value: string): any;
-  create(members: object): object;
-  flags(descriptor: EnumDescriptor): EnumDescriptor;
-};
+enum Numbers {
+  zero, // 0
+  one,  // 1
+  two   // 2
+}
 ```
 
-<!--
-# Grammar
+However, this behavior is contentious amongst some TC39 delegates and has been removed from this proposal. TypeScript
+will continue to support auto-initializiation due to its frequent use within the developer community, but would emit
+explicit initializers to JavaScript. It is possible that another form of auto-initializiation may be introduced in the
+future that could be utilized by both TypeScript and ECMAScript.
 
-```grammarkdown
-```
--->
+## Declaration Merging
 
-# Examples
+TypeScript (as of v5.8) allows `enum` declarations to merge with other `enum` (and `namespace`) declarations with the
+same name. This is not a desirable feature for ECMAScript enums and will not be supported. TypeScript is considering
+[deprecating this feature](https://github.com/microsoft/TypeScript/issues/54500#issuecomment-2770170732) in
+TypeScript 6.0.
 
-<!-- Examples of the proposal -->
+## Reverse Mapping
+
+TypeScript currently supports reverse-mapping enum values back to enum member names using `E[value]`, but only for
+`Number` values. This limitation is intended to avoid collisions for `String`-valued enum members that could potentially
+overwrite other members. While this information is invaluable for debugging, diagnostics, formatting, and serialization,
+it is far less frequently used compared to `enum` on the whole.
+
+To avoid this inconsistency, we propose instead propose introducing reverse mapping by way of the `@@iterator` built-in
+symbol:
 
 ```js
-enum Numbers { zero, one, two, three, }
+enum E {
+  A = 0,
+  B = "A",
+}
 
-typeof Numbers.zero === "number"
-Numbers.zero === 0
-Enum.getName(Numbers, 0) === "zero"
-Enum.parse(Numbers, "zero") === 0
+for (const [key, value] of E) {
+  console.log(`${key}: ${value}`);
+}
 
-// ... strings, ...
-enum HttpMethods of String { GET, PUT, POST, DELETE }
+// prints:
+//  A: 0
+//  B: A
+```
 
-typeof HttpMethods.GET === "string"
-HttpMethods.GET === "GET"
+If adopted, TypeScript would add support for `@@iterator` while eventually deprecating existing reverse mapping support.
 
-// ... booleans, ...
-enum Switch { on = true, off = false }
 
-typeof Switch.on === "boolean";
-Switch.on === true
+## `const enum`
 
-// ... symbols, ...
-enum AlphaBeta of Symbol { alpha, beta }
+TypeScript supports the concept of a `const enum` declaration, which is similar to a normal `enum` declaration except
+that enum values are inlined into their use sites. As this capability requires whole program knowledge, it is not a
+feature we intend to support in ECMAScript enums.
```

Comment:
This seems overly pessimistic.  Given the proposed JS enums are immutable, it would be good to assert that we have a goal of enabling optimising compilers (e.g. minifiers) to inline string/number values if they want.  That's a capability the spec would allow, not something that JS implementations need to care about.

Reply:
Implementations are free to inline enum values if possible. `const enum` is a TypeScript specific feature to do this at compile time. We are not suggesting bringing `const enum` to ECMAScript as it would (hopefully) be unnecessary.

Reply:
Ah I see. You are focussing specifically on distancing from the TS bit. That is fair to be explicit. Can we say this is a "yes and..." situation then?

Meaning after we say "there is no intent to introduce the TS const enum" we follow it up with "The reason this is unnecessary is because this proposal already has a goal of permitting tooling or engines to transparently inline the values."